### ISSUE TITLE
Adding prefer-for-of option breaks tslint

### ISSUE THREAD

**Issue by kemsky** [palantir/tslint#1884]
### Bug Report

- __TSLint version__: `4.1.0`
- __TypeScript version__: `2.1.4`
- __Running TSLint via__: gulp-tslint

#### TypeScript code being linted

very large application...

with `tslint.json` configuration:

```
"prefer-for-of": true
```

#### Actual behavior

Fails after adding `prefer-for-of`:

```
TypeError: Cannot read property 'onlyArrayReadAccess' of undefined
    at PreferForOfWalker.visitForStatement (C:\<project>\node_modules\tslint\lib\rules\preferForOfRule.js:73:33)
    at PreferForOfWalker.SyntaxWalker.visitNode (C:\<project>\node_modules\tslint\lib\language\walker\syntaxWalker.js:352:22)
    at C:\<project>\node_modules\tslint\lib\language\walker\syntaxWalker.js:517:63
    at visitEachNode (C:\<project>\node_modules\typescript\lib\typescript.js:13907:30)
    at Object.forEachChild (C:\<project>\node_modules\typescript\lib\typescript.js:14076:24)
    at PreferForOfWalker.SyntaxWalker.walkChildren (C:\<project>\node_modules\tslint\lib\language\walker\syntaxWalker.js:517:12)
    at PreferForOfWalker.SyntaxWalker.visitBlock (C:\<project>\node_modules\tslint\lib\language\walker\syntaxWalker.js:47:14)
    at PreferForOfWalker.SyntaxWalker.visitNode (C:\<project>\node_modules\tslint\lib\language\walker\syntaxWalker.js:289:22)
    at C:\<project>\node_modules\tslint\lib\language\walker\syntaxWalker.js:517:63
    at visitNode (C:\<project>\node_modules\typescript\lib\typescript.js:13895:20)

```

#### Expected behavior

Work as usual

**Comment by adidahiya** [comment]
Debugging this without a code sample will be difficult; we already run the rule in our test suite, so you may be using some syntax that wasn't accounted for in the rule implementation. Can you try to find a minimal repro?

**Comment by nchen63** [comment]
@kemsky I've committed a likely fix. Please reopen this issue if it is not fixed with #1926

**Comment by kemsky** [comment]
Version 4.3.1 works fine. Thank you!
